app/assets/javascripts/components/revisions/diff_viewer.jsx

```diff
@@ -53,20 +56,44 @@ const DiffViewer = createReactClass({
   },
 
   showDiff() {
-    this.setState({ showDiff: true });
+    this.setSelectedIndex(this.props.index);
+    this.fetchRevisionDetails();
+  },
+
+  fetchRevisionDetails() {
     if (!this.props.editors) {
       this.props.fetchArticleDetails();
     } else if (!this.state.fetched) {
       this.initiateDiffFetch(this.props);
     }
   },
 
+  shouldShowDiff(props) {
+    return props.selectedIndex === this.props.index;
+  },
+
   hideDiff() {
-    this.setState({ showDiff: false });
+    this.setSelectedIndex(-1);
+  },
+
+  showPreviousArticle() {
+    this.setSelectedIndex(this.props.index - 1);
+  },
+
+  showNextArticle() {
+    this.setSelectedIndex(this.props.index + 1);
+  },
+
+  isFirstArticle() {
+    return this.props.index === 0;
+  },
+
+  isLastArticle() {
+    return this.props.index === this.props.lastIndex - 1;
   },
 
   handleClickOutside() {
-    this.hideDiff();
+    // this.props.hideDiff();
```

Comment:
For handleClickOutside, it should either be reimplemented to only call `hideDiff` if it's currently open (because I assume interference between the multiple instances of this component is why it needed to be commented out), or this component shouldn't use the `OnClickOutside` high order component wrapper at all.

I think keeping the OnClickOutside behavior and just fixing it so that it works with this new functionality is slightly preferable, but I don't particularly mind removing it.